sdstore/cache2disk.go

```diff
@@ -1,21 +1,230 @@
 package sdstore
 
 import (
+	"encoding/json"
 	"fmt"
 	"github.com/otiai10/copy"
+	"io/ioutil"
+	"log"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strings"
 )
 
+// taken and modified from https://stackoverflow.com/questions/32482673/how-to-get-directory-total-size
+/*
+get directory size recursive in bytes
+param - path         		directory path
+return - size in bytes
+*/
+func getDirSizeInBytes(path string) int64 {
+	sizes := make(chan int64)
+	readSize := func(path string, file os.FileInfo, err error) error {
+		if err != nil || file == nil {
+			return nil
+		}
+		if !file.IsDir() {
+			sizes <- file.Size()
+		}
+		return nil
+	}
+
+	go func() {
+		_ = filepath.Walk(path, readSize)
+		close(sizes)
+	}()
+
+	size := int64(0)
+	for s := range sizes {
+		size += s
+	}
+
+	return size
+}
+
+/*
+compare md5 for source and destination directories
+param - src         		source directory
+param - dest			destination directory
+return - bytearray / error   	return md5 byte array, error - md5-same / md-chagned
+*/
+func checkMd5(src, dest string) ([]byte, error) {
+	var oldMd5 map[string]string
+	var newMd5 map[string]string
+
+	fmt.Println("start md5 check")
+	oldMd5FilePath := filepath.Join(filepath.Dir(dest), "md5.json")
+	oldMd5File, err := ioutil.ReadFile(oldMd5FilePath)
+	if err != nil {
+		oldMd5File = []byte("")
+		log.Printf("error: %v, not able to get md5.json from: %v \n", err, filepath.Dir(dest))
+	}
+	_ = json.Unmarshal(oldMd5File, &oldMd5)
+
+	if newMd5, err = MD5All(src); err != nil {
+		log.Printf("error: %v, not able to generate md5 for directory: %v \n", err, src)
+	}
+	md5Json, _ := json.Marshal(newMd5)
+
+	if reflect.DeepEqual(oldMd5, newMd5) {
+		return md5Json, fmt.Errorf("md5-same")
```

Comment:
this need not be forced into error, 2nd return value can be a boolean in this case.

Reply:
yeah thought of changing it to boolean, but left if as for now. anyway let me change it to bool